perses/app/setup_relative_calculation.py

```diff
@@ -739,7 +748,9 @@ def run(yaml_filename=None):
 
     _logger.info(f"Getting setup options from {yaml_filename}")
 
-    setup_options = getSetupOptions(yaml_filename)
+    # If a user doesn't pass in setup_options, generate them
+    if not setup_options:
+        setup_options = getSetupOptions(yaml_filename)
```

Comment:
I see an issue with this conditional, because you can end up shooting yourself in the foot rather easily when using Perses from the API, by calling the `run` function with any `setup_options` parameter. For example the following:

```python
from perses.app.setup_relative_calculation import getSetupOptions, run
run("my.yaml", {"phases": ["solvent", "complex"]})
```
This will end up in an error like `KeyError: 'trajectory_directory'` somewhere else in the code because not calling `getSetupOptions` will leave undefined many of the required options.

Reply:
Changing this, by forcing to always call `getSetupOptions`, should also help with the issue commented in `cli.py` in this review.

Reply:
:100: agree but I think this is outside what this PR does. As I understand it, no one should use perses like this 

```
from perses.app.setup_relative_calculation import getSetupOptions, run
run("my.yaml", {"phases": ["solvent", "complex"]})
```

Either you make objects in the python API or use the CLI, but don't use the `setup_relative*.py` like that since they really are not setup to do that. I only did that because I needed a way to override options _after_ `getSetupOptions` does its thing.